game-backend/src/game-engine/intent/AttackIntent.ts

```diff
@@ -0,0 +1,47 @@
+import { State, UserContext, Unit, ObjectId } from '@darwin/types';
+import { Intent } from './Intent';
+import { getUnit } from '../../helper/gameObjects';
+import produce from '../../helper/produce';
+
+export const ATTACK_RANGE = 1;
+export const HIT_DAMAGE = 10;
+
+/**
+ * Represents the intent to attack an enemy.
+ */
+export default class AttackIntent implements Intent {
+  cost = 2;
+
+  private targetUnitId: ObjectId;
+
+  constructor(targetUnit: Unit) {
+    this.targetUnitId = targetUnit.id;
+  }
+
+  private static isTargetInReach(unit: Unit, targetUnit: Unit): boolean {
+    const xDiff = Math.abs(unit.position.x - targetUnit.position.x);
+    const yDiff = Math.abs(unit.position.y - targetUnit.position.y);
+    const diff = Math.sqrt(xDiff ** 2 + yDiff ** 2);
+
+    return diff <= ATTACK_RANGE;
+  }
+
+  private static attack(targetUnit: Unit, state: State): State {
+    return produce(state, draft => {
+      const defendingUnit = getUnit(draft, targetUnit.id);
+      defendingUnit.health -= HIT_DAMAGE;
```

Comment:
Given:
- an enemy unit with 5 health is on the **left side** of your own unit
- you attack the enemy unit
- the enemy unit has -5 health and is therefore dead. It shouldn't be on the field anymore
- the own unit wants to move **left** - this is not possible as the dead enemy unit is still on the field

Or do I miss something?

Reply:
Yes, you are right. We need to remove the unit like we do when it starves to death. It should be the same dying behaviour after it gets to <= 0 HP.